app/code/Magento/CustomerGraphQl/Model/Resolver/DeleteCustomer.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CustomerGraphQl\Model\Resolver;
+
+use Magento\CustomerGraphQl\Model\Customer\DeleteCustomer as DeleteCustomerModel;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\Registry;
+use Magento\GraphQl\Model\Query\ContextInterface;
+
+/**
+ * Delete customer, used for GraphQL request processing.
+ */
+class DeleteCustomer implements ResolverInterface
+{
+    /**
+     * @var DeleteCustomerModel
+     */
+    private $deleteCustomer;
+
+    /** @var Registry */
+    private $registry;
+
+    /**
+     * @param DeleteCustomerModel $deleteCustomer
+     * @param Registry $registry
+     */
+    public function __construct(
+        DeleteCustomerModel $deleteCustomer,
+        Registry $registry
+    ) {
+        $this->deleteCustomer = $deleteCustomer;
+        $this->registry =$registry;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        /** @var ContextInterface $context */
+        if (false === $context->getExtensionAttributes()->getIsCustomer()) {
+            throw new GraphQlAuthorizationException(__('The current customer isn\'t authorized.'));
+        }
+
+        $isSecure = $this->registry->registry('isSecureArea');
+
+        $this->registry->unregister('isSecureArea');
+        $this->registry->register('isSecureArea', true);
```

Comment:
The check is located in `magento/lib/internal/Magento/Framework/Model/ActionValidator/RemoveAction.php`
```
 public function isAllowed(AbstractModel $model)
    {
        $isAllowed = true;

        if ($this->registry->registry('isSecureArea')) {
            $isAllowed = true;
        } elseif (in_array($this->getBaseClassName($model), $this->protectedModels)) {
            $isAllowed = false;
        }
        return $isAllowed;
    }
```
ProtectedModels include the customer model too. In the file magento/app/code/Magento/Customer/etc/di.xml
```
<type name="Magento\Framework\Model\ActionValidator\RemoveAction">
        <arguments>
            <argument name="protectedModels" xsi:type="array">
                <item name="customer" xsi:type="string">Magento\Customer\Model\Customer</item>
            </argument>
        </arguments>
    </type>
```

Reply:
To avoid deprecated `\Magento\Framework\Registry`, we can remove the Customer model from protected models in general or copy all protected models to graphQL/di.xml for all magento modules except the Customer one.